llvm-spirv/lib/SPIRV/CMakeLists.txt

```diff
@@ -1,54 +1,107 @@
-add_llvm_library(LLVMSPIRVLib
-  LLVMSPIRVOpts.cpp
-  LLVMToSPIRVDbgTran.cpp
-  Mangler/FunctionDescriptor.cpp
-  Mangler/Mangler.cpp
-  Mangler/ManglingUtils.cpp
-  Mangler/ParameterType.cpp
-  OCLToSPIRV.cpp
-  OCLTypeToSPIRV.cpp
-  OCLUtil.cpp
-  VectorComputeUtil.cpp
-  SPIRVLowerBitCastToNonStandardType.cpp
-  SPIRVLowerBool.cpp
-  SPIRVLowerConstExpr.cpp
-  SPIRVLowerMemmove.cpp
-  SPIRVLowerOCLBlocks.cpp
-  SPIRVLowerSaddWithOverflow.cpp
-  SPIRVReader.cpp
-  SPIRVRegularizeLLVM.cpp
-  SPIRVToLLVMDbgTran.cpp
-  SPIRVToOCL.cpp
-  SPIRVToOCL12.cpp
-  SPIRVToOCL20.cpp
-  SPIRVUtil.cpp
-  SPIRVWriter.cpp
-  SPIRVWriterPass.cpp
-  PreprocessMetadata.cpp
-  libSPIRV/SPIRVBasicBlock.cpp
-  libSPIRV/SPIRVDebug.cpp
-  libSPIRV/SPIRVDecorate.cpp
-  libSPIRV/SPIRVEntry.cpp
-  libSPIRV/SPIRVFunction.cpp
-  libSPIRV/SPIRVInstruction.cpp
-  libSPIRV/SPIRVModule.cpp
-  libSPIRV/SPIRVStream.cpp
-  libSPIRV/SPIRVType.cpp
-  libSPIRV/SPIRVValue.cpp
-  LINK_COMPONENTS
-    Analysis
-    BitWriter
-    CodeGen
-    Core
-    Demangle
-    IRReader
-    Linker
-    Passes
-    Support
-    TransformUtils
+if(LLVM_LINK_LLVM_DYLIB)
```

Comment:
@fineg74, please, commit LLVM-SPIRV-Translator changes to https://github.com/KhronosGroup/SPIRV-LLVM-Translator/ repository. We regularly pull changes from Khronos repository here.

I thought that llvm-spirv tool build is fixed by https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/458. Has it been broken again?

Reply:
It seems https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1543 is the PR mentioned and it should land first.
@fineg74, please add detailed rationale in PR description there. Not sure if testing fails due to infrastructure temporary problems, you can restart if it was. I guess failed testing and no description are the reasons patch stuck for a while.

Reply:
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1543 is merged, so I expect no changes in llvm-spirv/* files.

I agree with Pavel that description should be improved. It's not clear why we need to change customize cmake files here. I expected `add_llvm_library` to handle linking with all intrinsics. I assume we are building all libraries from sources, so compilation and linking with intrinsics_gen should be no different than any other intrinsics like X86 or NVPTX intrinsics, which do not require this changes. Could you elaborate in the description the difference with regular intrinsics, which makes us customize cmake files for these tools, please?